client/src/modules/dashboard/Events/components/EventCancelButton.tsx

```diff
@@ -0,0 +1,53 @@
+import { Button, ButtonProps } from '@chakra-ui/react';
+import { useConfirm } from 'chakra-confirm';
+import React, { useMemo } from 'react';
+import { Event, useCancelEventMutation } from '../../../../generated/graphql';
+import { EVENT, EVENTS } from '../graphql/queries';
+
+interface EventCancelButtonProps extends ButtonProps {
+  isFullWidth?: boolean;
+  buttonText: string;
+  event: Pick<Event, 'id' | 'canceled'>;
+}
+
+const EventCancelButton: React.FC<EventCancelButtonProps> = (props) => {
+  const { isFullWidth, buttonText, event, ...rest } = props;
+
+  const [cancel] = useCancelEventMutation();
+
+  const confirmCancel = useConfirm({
+    title: 'Are you sure you want to cancel this',
+    body: 'Canceling this will send emails to everyone who RSVPd',
+    buttonColor: 'orange',
+  });
+
+  const data = useMemo(
+    () => ({
+      variables: { id: event.id },
+      refetchQueries: [
+        { query: EVENT, variables: { id: event.id } },
+        { query: EVENTS },
+      ],
+    }),
+    [event],
+  );
```

Comment:
Right, I followed this from elsewhere. If I understand correctly, you suggest to simply do instead:

```suggestion
  const data = {
    variables: { id: event.id },
    refetchQueries: [
      { query: EVENT, variables: { id: event.id } },
      { query: EVENTS },
    ],
  };
```

Reply:
Yep, exactly that. Unless I'm missing something, that's essentially the same, but simpler.